service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/NoticeOfChangeEventHandlerTest.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.hmcts.reform.fpl.handlers;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import uk.gov.hmcts.reform.fpl.events.NoticeOfChangeEvent;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.RespondentSolicitor;
+import uk.gov.hmcts.reform.fpl.model.notify.noticeofchange.NoticeOfChangeRespondentSolicitorTemplate;
+import uk.gov.hmcts.reform.fpl.service.email.NotificationService;
+import uk.gov.hmcts.reform.fpl.service.email.content.NoticeOfChangeContentProvider;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.reform.fpl.NotifyTemplates.NOTICE_OF_CHANGE_FORMER_REPRESENTATIVE;
+import static uk.gov.hmcts.reform.fpl.NotifyTemplates.NOTICE_OF_CHANGE_NEW_REPRESENTATIVE;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.caseData;
+
+@ExtendWith(MockitoExtension.class)
+@SpringBootTest(classes = {NoticeOfChangeEventHandler.class})
+public class NoticeOfChangeEventHandlerTest {
+
+    private static final String expectedEmail = "test@test.com";
+    private static final RespondentSolicitor solicitor = RespondentSolicitor.builder().email(expectedEmail).build();
+    private static final  CaseData caseData = caseData();
+    private static final NoticeOfChangeRespondentSolicitorTemplate expectedTemplate =
+        NoticeOfChangeRespondentSolicitorTemplate.builder()
+            .build();
+
+    @MockBean
+    private NotificationService notificationService;
+
+    @MockBean
+    private NoticeOfChangeContentProvider noticeOfChangeContentProvider;
+
+    @Autowired
+    private NoticeOfChangeEventHandler underTest;
+
+    @Test
+    void shouldSendEmailToSolicitorAccessGranted() {
+        given(noticeOfChangeContentProvider.buildRespondentSolicitorAccessGrantedNotification(caseData, solicitor))
+            .willReturn(expectedTemplate);
+
+        underTest.notifySolicitorAccessGranted(new NoticeOfChangeEvent(caseData, solicitor, solicitor));
```

Comment:
this is not right to pass same solicitor as new one and old one. 

This test does not prove that email is sent to new solicitor and not old one as both have same email